lgc/patch/PatchResourceCollect.cpp

```diff
@@ -721,32 +722,61 @@ bool PatchResourceCollect::checkGsOnChipValidity() {
         }
 
         esVertsPerSubgroup = gsPrimsPerSubgroup * vertsPerPrimitive;
+        if (hasTs)
+          esVertsPerSubgroup = std::min(esVertsPerSubgroup, OptimalVerticesPerPrimitiveForTess * gsPrimsPerSubgroup);
       } else {
         // If GS is not present, instance count must be 1
         assert(gsInstanceCount == 1);
       }
 
-      // Make sure that we have at least one primitive
-      assert(gsPrimsPerSubgroup >= 1);
+      unsigned expectedEsLdsSize = esVertsPerSubgroup * esGsRingItemSize + esExtraLdsSize;
+      unsigned expectedGsLdsSize = gsPrimsPerSubgroup * gsInstanceCount * gsVsRingItemSize + gsExtraLdsSize;
 
-      // HW has restriction on NGG + Tessellation where gsPrimsPerSubgroup must be >= 3.
-      assert(!hasTs || gsPrimsPerSubgroup >= 3);
+      const unsigned ldsSizeDwordGranularity =
+          1u << m_pipelineState->getTargetInfo().getGpuProperty().ldsSizeDwordGranularityShift;
+      unsigned ldsSizeDwords = alignTo(expectedEsLdsSize + expectedGsLdsSize, ldsSizeDwordGranularity);
 
-      unsigned expectedEsLdsSize = esVertsPerSubgroup * esGsRingItemSize + esExtraLdsSize;
-      const unsigned expectedGsLdsSize = gsPrimsPerSubgroup * gsInstanceCount * gsVsRingItemSize + gsExtraLdsSize;
+      static const unsigned MaxHwGsLdsSizeDwords = 16384;
```

Comment:
please get the value from gfx property.

Reply:
Sorry. I miss this. I will make another PR to correct this.